uritemplate/validator.py

```diff
@@ -0,0 +1,58 @@
+"""Module containing all the validation logic for uritemplate."""
+from uritemplate import exceptions as exc
+
+
+class Validator(object):
+    """Provide configurable validation of URITemplate objects.
+
+    .. versionadded:: 3.1.0
+
+    .. code-block::
+
+        from uritemplate import URITemplate, Validator
+
+
+    """
+
+    def __init__(self):
+        """Initialize our validator."""
+        self.enforcing_unbalanced_braces = True
+
+    def allow_unbalanced_braces(self):
+        """Allow a template to have unbalanced braces.
+
+        .. versionadded:: 3.1.0
+
+        Returns the validator instance.
+        """
+        self.enforcing_unbalanced_braces = False
+        return self
```

Comment:
I take it you intentionally like chaining, even in cases where the instance itself has been mutated?